### ISSUE TITLE
Coverage GUI: Show partial affects Coverpoints, not groups

### ISSUE THREAD

**Issue by nrother** [amiq-consulting/fc4sc#16]
In the HTML GUI, when disabling "Show partial" cover**points** with less then 100% coverage are filtered, but not the complete cover**group** (if the group contains at least one coverpoint with 100% coverage). The filtered coverpoint of removed from the covergroup completely, yielding the strange effect, that the covergroups shows up with 100% coverage, but with a yellow font color.

In my opinion, the "Show partial" filter should by applied to complete covergroups, not individual coverpoints.

**Comment by nrother** [comment]
The same seems to apply for the other filters, these also work on coverpoints, not groups.

Also there seems to be some issues with re-opening the same file without a page reload, which sometime causes duplicate to show up in the GUI after using filters, but I was not yet able to reproduce this completely.

**Comment by amiq-consulting** [comment]
Hi nrother,

The behaviour of the buttons was strange indeed and I agree that it is more intuitive to filter by covergroup rather than coverpoint.
I have implemented this behavior, available on the main branch here: [https://github.com/amiq-consulting/fc4sc/commit/2550a01aa2e34f77a86f43e9babdf3f1cf24806c](https://github.com/amiq-consulting/fc4sc/commit/2550a01aa2e34f77a86f43e9babdf3f1cf24806c)

Regarding your second issue: Yes, I have also noticed that while testing the new filter functionality. The commit above should fix that (assuming we did experience the same issue).

Thank you.
Best Regards,
Dragos Dospinescu
